### ISSUE TITLE
Seed Nodes Unable to Start Cluster Depending on Ordering of HOCON Seed Node List

### ISSUE THREAD

**Issue by nrgonzalez777** [akkadotnet/akka.net#1719]
Hello,

I've included a very simple test application that showcases the behavior described in the title. I have a HOCON configuration that defines two seed nodes: localhost:8081 and localhost:8082. If I start up the console app using 8081 first, everything works fine. 8081 starts up, tries to connect to 8082 and fails, then goes on to create a cluster. Unfortunately this same behavior doesn't occur if I set the application up to use 8082 (which is the second value in the seed nodes list). 8082 will sit there and try to connect to 8081 forever, without coming to the conclusion that it is also a seed node and it can start up its own cluster. I would expect the behavior to be that it would eventually try the second entry in the seed node list and successfully form a cluster by itself. This is definitely an issue with the ordering of the seed node list, because if simply reverse the entries so that 8082 is the first value, 8082 works fine but 8081 will not start up properly.

Please let me know if you need any additional information or help getting the problem to repro.

[Akka.Cluster.Test.zip](https://github.com/akkadotnet/akka.net/files/138557/Akka.Cluster.Test.zip)

**Comment by shivanan** [comment]
Seems like this is intended behaviour?

From the [Java Akka Documentation](http://doc.akka.io/docs/akka/2.4.1/scala/cluster-usage.html) :

> The seed nodes can be started in any order and it is not necessary to have all seed nodes running, but the node configured as the first element in the seed-nodes configuration list must be started when initially starting a cluster, otherwise the other seed-nodes will not become initialized and no other node can join the cluster. The reason for the special first seed node is to avoid forming separated islands when starting from an empty cluster. It is quickest to start all configured seed nodes at the same time (order doesn't matter), otherwise it can take up to the configured seed-node-timeout until the nodes can join.

**Comment by Aaronontheweb** [comment]
Sure looks that way - I'll update the Cluster docs.

**Comment by ZoolWay** [comment]
Unfortunately this does not seem to have been added to the Akka.NET documentation.

At first I was afraid that this would cause problems for new nodes to join after the first special seed node crashed. But I tested it and other nodes could join the cluster even after the first-specia-seed-node crashed. Only requirement is that is must have been existing and have been connected to other seed nodes when the cluster started.
